kickstart/provision.erb

```diff
@@ -30,7 +30,8 @@ This template accepts the following parameters:
   realm_compatible = (@host.operatingsystem.name == 'Fedora' && os_major >= 20) || (rhel_compatible && os_major >= 7)
   # safemode renderer does not support unary negation
   pm_set = @host.puppetmaster.empty? ? false : true
-  proxy_string = @host.params['http-proxy'] ? " --proxy=http://#{@host.params['http-proxy']}:#{@host.params['http-proxy-port']}" : ''
+  proxy_uri = @host.params['http-proxy'] ? "http://#{@host.params['http-proxy']}:#{@host.params['http-proxy-port']}" : ''
+  proxy_string = proxy_uri ? " --proxy=#{proxy_uri}" : ''
```

Comment:
proxy_uri is always trueish, because the empty string assigned on line 33 is trueish, so it's generating `--proxy=` in the kickstart